tests/run_tests.py

```diff
@@ -351,33 +399,56 @@ def run_system(config_file, vic_exe, test_data_dir, out_dir, driver):
         error_message = ''
 
         try:
-            # Run the VIC simulation
-            returncode = vic_exe.run(test_global_file, logdir=dirs['logs'],
-                                     **run_kwargs)
-            test_complete = True
+            if 'restart' in test_dict:
+                for j, test_global_file in enumerate(list_test_global_file):
+                    returncode = vic_exe.run(test_global_file,
+                                             logdir=dirs['logs'])
+                    # Check return code
+                    check_returncode(returncode,
+                                     test_dict.pop('expected_retval', 0))
+            else:
+                returncode = vic_exe.run(test_global_file, logdir=dirs['logs'],
+                                         **run_kwargs)
+                # Check return code
+                check_returncode(returncode,
+                                 test_dict.pop('expected_retval', 0))
 
-            # Check return code
-            check_returncode(returncode, test_dict.pop('expected_retval', 0))
+            test_complete = True
 
             # check output files (different tests depending on driver)
             if test_dict['check']:
```

Comment:
I think this should be `if 'check' in test_dict:`  I know you didn't write this but it would be worth fixing.